src/appState.ts

```diff
@@ -74,6 +74,7 @@ export const getDefaultAppState = (): Omit<
     zenModeEnabled: false,
     zoom: { value: 1 as NormalizedZoomValue, translation: { x: 0, y: 0 } },
     viewModeEnabled: false,
+    customName: false,
```

Comment:
why do we need a new state ? lets use the `name` only

Reply:
I was going to comment regarding this. I was using this to find out if the `name` is either passed as a prop or if it is the default name from state in the action so that I can disable editing in the input accordingly. Is there any way I can do this without relying on this?

Reply:
you can directly check whether the prop name is passed, if passed the value of state can be set accordingly. If you are talking about deciding when to disable input inside export modal then you can pass a prop `isNameEditable` as true when `name` prop is `undefined` to the `ProjectName`, similar to how we don't show `zenModeEnabled`/ `gridModeEnabled` in context menu when those props are passed or theme switch button when theme is passed.

Reply:
I understand the example. In this case, since the `ProjectName` component is inside an action, I'm trying figure out how exactly I should pass a prop to that.